### PR TITLE
[IMP] website: introduce `s_images_constellation` snippet

### PR DESCRIPTION
**Pull request by chgo-odoo**
This PR introduces a new snippet named `s_images_constellation`.

- requires : https://github.com/odoo/design-themes/pull/879

task-4105366
part of task-4077427

---
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

File: addons/website/views/snippets/s_images_constellation.xml [hunk 1]
```diff
@@ -0,0 +1,31 @@
+<?xml version="1.0" encoding="utf-8"?>
+<odoo>
+
+<template id="s_images_constellation" name="Images constellation">
+    <section class="s_images_constellation o_cc o_cc5 pt48 pb48">
+        <div class="container">
+            <div class="row o_grid_mode" data-row-count="23">
+                <div class="o_grid_item o_grid_item_image g-col-lg-3 g-height-4 col-lg-4 o_snippet_mobile_invisible d-none d-lg-block" style="z-index: 1; grid-area: 3 / 6 / 7 / 9; --grid-item-padding-x: 0px; --grid-item-padding-y: 0px;">
```
Comment (xlu-odoo) [inline addons/website/views/snippets/s_images_constellation.xml:8]: Even if you don't plan to let user use col, i'd correct `col-lg-*` to match the `g-col-lg-*` for all grid items;

Comment (xlu-odoo) [inline addons/website/views/snippets/s_images_constellation.xml:8]: nitpicking; `o_snippet_mobile_invisible` here and on 6th item to be placed at the end, so coherent with the other ones.

```suggestion
                <div class="o_grid_item o_grid_item_image g-col-lg-3 g-height-4 col-lg-4 d-none d-lg-block o_snippet_mobile_invisible " style="z-index: 1; grid-area: 3 / 6 / 7 / 9; --grid-item-padding-x: 0px; --grid-item-padding-y: 0px;">
```

File: addons/website/views/snippets/s_images_constellation.xml [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+                </div>
+                <div class="o_grid_item o_grid_item_image g-col-lg-4 g-height-6 col-12 col-lg-6 offset-lg-4" style="z-index: 2; grid-area: 7 / 1 / 13 / 5; --grid-item-padding-x: 16px; --grid-item-padding-y: 0px;">
+                    <img class="img img-fluid mx-auto rounded" src="/web/image/website.s_picture_default_image" alt=""/>
+                </div>
+                <div class="o_grid_item o_grid_item_image g-col-lg-4 g-height-6 col-lg-5 offset-lg-1 d-none d-lg-block o_snippet_mobile_invisible" style="z-index: 3; grid-area: 7 / 9 / 13 / 13; --grid-item-padding-x: 0px; --grid-item-padding-y: 24px;">
+                    <img class="img img-fluid mx-auto rounded" alt=""  data-shape="web_editor/geometric/geo_sonar" data-file-name="s_images_constellation_default_image.jpg" data-original-mimetype="image/jpeg/" src="/web_editor/image_shape/website.s_images_constellation_default_image/web_editor/geometric/geo_sonar.svg"/>
+                </div>
+                <div class="justify-content-center o_grid_item g-height-8 g-col-lg-6 col-12 col-lg-6" style="z-index: 6; grid-area: 9 / 5 / 17 / 11; --grid-item-padding-x: 16px; --grid-item-padding-y: 16px;">
```
Comment (xlu-odoo) [inline addons/website/views/snippets/s_images_constellation.xml:17]:
```suggestion
                <div class="o_grid_item g-col-lg-6 g-height-8 col-lg-6 col-12 justify-content-center" style="z-index: 6; grid-area: 9 / 5 / 17 / 11; --grid-item-padding-x: 16px; --grid-item-padding-y: 16px;">
```

File: addons/website/views/snippets/s_images_constellation.xml [hunk 3]
```diff
@@ -0,0 +1,31 @@
+<?xml version="1.0" encoding="utf-8"?>
+<odoo>
+
+<template id="s_images_constellation" name="Images constellation">
+    <section class="s_images_constellation o_cc o_cc5 pt48 pb48">
+        <div class="container">
+            <div class="row o_grid_mode" data-row-count="23">
+                <div class="o_grid_item o_grid_item_image g-col-lg-3 g-height-4 col-lg-4 o_snippet_mobile_invisible d-none d-lg-block" style="z-index: 1; grid-area: 3 / 6 / 7 / 9; --grid-item-padding-x: 0px; --grid-item-padding-y: 0px;">
+                    <img class="img img-fluid mx-auto rounded" src="/web/image/website.s_carousel_default_image_2" alt=""/>
```
Comment (xlu-odoo) [inline addons/website/views/snippets/s_images_constellation.xml:9]: Wondering if mx-auto is important here. wdyt?

File: addons/website/views/snippets/snippets.xml [hunk 4]
```diff
@@ -661,7 +664,7 @@
     <div data-js="layout_column"
         data-selector="section, section.s_carousel_wrapper .carousel-item"
         data-target="> *:has(> .row), > .s_allow_columns"
-        data-exclude=".s_masonry_block, .s_features_grid, .s_media_list, .s_table_of_content, .s_process_steps, .s_image_gallery, .s_timeline">
+        data-exclude=".s_masonry_block, .s_features_grid, .s_media_list, .s_table_of_content, .s_process_steps, .s_image_gallery, .s_timeline, .s_images_constellation">
```
Comment (xlu-odoo) [inline addons/website/views/snippets/snippets.xml:667]: Since you're removing the ability to switch to col, it is still important to be able to add new images/text/button. so a few lines above, add the snippet in the data-selector.

```xml
 <!-- Grid only -->
<div data-js="layout_column"
    data-selector="section.s_masonry_block, section.s_images_constellation"
    data-target="> *:has(> .row)">
	<t t-call="website.grid_layout_options"/>
</div>
```

File: addons/website/views/snippets/s_images_constellation.xml [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+                </div>
+                <div class="o_grid_item o_grid_item_image g-col-lg-4 g-height-6 col-12 col-lg-6 offset-lg-4" style="z-index: 2; grid-area: 7 / 1 / 13 / 5; --grid-item-padding-x: 16px; --grid-item-padding-y: 0px;">
+                    <img class="img img-fluid mx-auto rounded" src="/web/image/website.s_picture_default_image" alt=""/>
+                </div>
+                <div class="o_grid_item o_grid_item_image g-col-lg-4 g-height-6 col-lg-5 offset-lg-1 d-none d-lg-block o_snippet_mobile_invisible" style="z-index: 3; grid-area: 7 / 9 / 13 / 13; --grid-item-padding-x: 0px; --grid-item-padding-y: 24px;">
```
Comment (xlu-odoo) [inline addons/website/views/snippets/s_images_constellation.xml:14]:
```suggestion
                <div class="o_grid_item o_grid_item_image g-col-lg-4 g-height-6 col-lg-4 d-none d-lg-block o_snippet_mobile_invisible" style="z-index: 3; grid-area: 7 / 9 / 13 / 13; --grid-item-padding-x: 0px; --grid-item-padding-y: 24px;">
```

File: addons/website/views/snippets/s_images_constellation.xml [hunk 6]
```diff
@@ -0,0 +1,31 @@
...
+<odoo>
+
+<template id="s_images_constellation" name="Images constellation">
+    <section class="s_images_constellation o_cc o_cc5 pt48 pb48">
+        <div class="container">
+            <div class="row o_grid_mode" data-row-count="23">
+                <div class="o_grid_item o_grid_item_image g-col-lg-3 g-height-4 col-lg-4 o_snippet_mobile_invisible d-none d-lg-block" style="z-index: 1; grid-area: 3 / 6 / 7 / 9; --grid-item-padding-x: 0px; --grid-item-padding-y: 0px;">
+                    <img class="img img-fluid mx-auto rounded" src="/web/image/website.s_carousel_default_image_2" alt=""/>
+                </div>
+                <div class="o_grid_item o_grid_item_image g-col-lg-4 g-height-6 col-12 col-lg-6 offset-lg-4" style="z-index: 2; grid-area: 7 / 1 / 13 / 5; --grid-item-padding-x: 16px; --grid-item-padding-y: 0px;">
```
Comment (xlu-odoo) [inline addons/website/views/snippets/s_images_constellation.xml:11]:
```suggestion
                <div class="o_grid_item o_grid_item_image g-col-lg-4 g-height-6 col-lg-4 col-12" style="z-index: 2; grid-area: 7 / 1 / 13 / 5; --grid-item-padding-x: 16px; --grid-item-padding-y: 0px;">
```

File: addons/website/views/snippets/s_images_constellation.xml [hunk 7]
```diff
@@ -0,0 +1,31 @@
...
+                    <img class="img img-fluid mx-auto rounded" alt=""  data-shape="web_editor/geometric/geo_sonar" data-file-name="s_images_constellation_default_image.jpg" data-original-mimetype="image/jpeg/" src="/web_editor/image_shape/website.s_images_constellation_default_image/web_editor/geometric/geo_sonar.svg"/>
+                </div>
+                <div class="justify-content-center o_grid_item g-height-8 g-col-lg-6 col-12 col-lg-6" style="z-index: 6; grid-area: 9 / 5 / 17 / 11; --grid-item-padding-x: 16px; --grid-item-padding-y: 16px;">
+                    <h2 class="display-3" style="text-align: right;">A constellation of amazing solutions tailored for your needs</h2>
+                </div>
+                <div class="o_grid_item o_grid_item_image g-col-lg-4 g-height-7 col-lg-4 offset-lg-8 d-none d-lg-block o_snippet_mobile_invisible" style="z-index: 4; grid-area: 15 / 2 / 22 / 6; --grid-item-padding-x: 0px; --grid-item-padding-y: 0px;">
```
Comment (xlu-odoo) [inline addons/website/views/snippets/s_images_constellation.xml:20]:
```suggestion
                <div class="o_grid_item o_grid_item_image g-col-lg-4 g-height-7 col-lg-4 d-none d-lg-block o_snippet_mobile_invisible" style="z-index: 4; grid-area: 15 / 2 / 22 / 6; --grid-item-padding-x: 0px; --grid-item-padding-y: 0px;">
```

File: addons/website/views/snippets/s_images_constellation.xml [hunk 8]
```diff
@@ -0,0 +1,31 @@
...
+                    <img class="img img-fluid mx-auto rounded" alt="" data-shape="web_editor/composite/composite_double_pill" data-file-name="s_images_constellation_default_image_1.jpg" data-original-mimetype="image/jpeg/" src="/web_editor/image_shape/website.s_images_constellation_default_image_1/web_editor/composite/composite_double_pill.svg"/>
+                </div>
+                <div class="o_grid_item o_grid_item_image g-col-lg-5 g-height-9 col-lg-4 o_snippet_mobile_invisible d-none d-lg-block" style="z-index: 5; grid-area: 15 / 8 / 24 / 13; --grid-item-padding-x: 0px; --grid-item-padding-y: 0px;">
```
Comment (xlu-odoo) [inline addons/website/views/snippets/s_images_constellation.xml:23]:
```suggestion
                <div class="o_grid_item o_grid_item_image g-col-lg-5 g-height-9 col-lg-5 o_snippet_mobile_invisible d-none d-lg-block" style="z-index: 5; grid-area: 15 / 8 / 24 / 13; --grid-item-padding-x: 0px; --grid-item-padding-y: 0px;">
```

**Review by xlu-odoo — CHANGES_REQUESTED** [review]
Snippet looks good, nice work. 🚀 
Some comments in the code

Also, nitpicking, snippets name in PR title inside ``

**Review by xlu-odoo — APPROVED** [review]

File: addons/website/views/snippets/s_images_constellation.xml [hunk 9]
```diff
@@ -0,0 +1,31 @@
...
+                </div>
+                <div class="o_grid_item o_grid_item_image g-col-lg-4 g-height-6 col-12 col-lg-4" style="z-index: 2; grid-area: 7 / 1 / 13 / 5; --grid-item-padding-x: 16px; --grid-item-padding-y: 0px;">
+                    <img class="img img-fluid mx-auto rounded" src="/web/image/website.s_picture_default_image" alt=""/>
+                </div>
+                <div class="o_grid_item o_grid_item_image g-col-lg-4 g-height-6 col-lg-4 d-none d-lg-block o_snippet_mobile_invisible" style="z-index: 3; grid-area: 7 / 9 / 13 / 13; --grid-item-padding-x: 0px; --grid-item-padding-y: 24px;">
+                    <img class="img img-fluid mx-auto rounded" alt=""  data-shape="web_editor/geometric/geo_sonar" data-file-name="s_images_constellation_default_image.jpg" data-original-mimetype="image/jpeg/" src="/web_editor/image_shape/website.s_images_constellation_default_image/web_editor/geometric/geo_sonar.svg"/>
+                </div>
+                <div class="o_grid_item g-height-8 g-col-lg-6 col-12 col-lg-6 justify-content-center" style="z-index: 6; grid-area: 9 / 5 / 17 / 11; --grid-item-padding-x: 16px; --grid-item-padding-y: 16px;">
```
Comment (bvr-odoo) [inline addons/website/views/snippets/s_images_constellation.xml:17]: Why did you add `justify-content-center` ? I don't think the editor can add this class, and it seems unnecessary.

**Review by bvr-odoo — CHANGES_REQUESTED** [review]
Hello @chgo-odoo 
Here is my review, thanks

**Comment by bvr-odoo** [comment]
@chgo-odoo don't forget this one :wink:

File: addons/website/views/snippets/s_images_constellation.xml (hunk 9 above)
Comment (chgo-odoo) [reply addons/website/views/snippets/s_images_constellation.xml:17]: left-over indeed 👍

**Review by bvr-odoo — APPROVED** [review]

**Comment by bvr-odoo** [comment]
Commit(s) moved into
- https://github.com/odoo/odoo/pull/180736
